sno/utils.py

```diff
@@ -0,0 +1,38 @@
+import functools
+from itertools import islice
+
+
+def ungenerator(cast_function):
```

Comment:
isn't this just `dict(mygenerator())` / `list(mygenerator())`?

Reply:
yes, but it's a decorator that applies that to a generator-returning function

Reply:
![](https://media0.giphy.com/media/RFYMmNb3j60BG/giphy.gif)

Reply:
I got britta'd.

Are you saying this isn't useful? Or could be implemented simpler?

It's just a nice sugar to avoid having to build up a dict/list in the function and then return it. The decorator can aggregate rather than the function code having to.

Reply:
I don’t really get it, but you’re very capable so it must be useful 😀👍